gno.land/pkg/gnoland/param.go

```diff
@@ -0,0 +1,121 @@
+package gnoland
+
+import (
+	"encoding/hex"
+	"errors"
+	"fmt"
+	"strconv"
+	"strings"
+
+	"github.com/gnolang/gno/tm2/pkg/sdk"
+	"github.com/gnolang/gno/tm2/pkg/sdk/params"
+)
+
+type Param struct {
+	key   string
+	kind  string
+	value interface{}
+}
+
+func (p Param) Verify() error {
+	// XXX: validate
+	return nil
+}
+
+const (
+	ParamKindString = "string"
+	ParamKindInt64  = "int64"
+	ParamKindUint64 = "uint64"
+	ParamKindBool   = "bool"
+	ParamKindBytes  = "bytes"
+)
+
+func (p *Param) Parse(entry string) error {
+	parts := strings.SplitN(strings.TrimSpace(entry), "=", 2) // <key>.<kind>=<value>
+	if len(parts) != 2 {
+		return fmt.Errorf("malformed entry: %q", entry)
+	}
+
+	keyWithKind := parts[0]
+	rawValue := parts[1]
+	p.kind = keyWithKind[strings.LastIndex(keyWithKind, ".")+1:]
+	p.key = strings.TrimSuffix(keyWithKind, "."+p.kind)
```

Comment:
Small improvement, we can avoid the double `strings.TrimSuffix`

```suggestion
	var (
	    keyWithKind = parts[0]
    	rawValue	= parts[1]
	)

    dotIndex := strings.LastIndex(keyWithKind, ".")
    if dotIndex == -1 {
        return fmt.Errorf("malformed key and kind in entry: %q", entry)
    }

    p.key = keyWithKind[:dotIndex]
    p.kind = keyWithKind[dotIndex+1:]
```